File: Application/fingerprint.py
# -*- coding: utf-8 -*-

# Create a hash "fingerprint" of a file
from hashlib import md5

# Create a unique fingerprint of a file
def fingerprint(f):
    block = 65536
    fingerprint = md5()
    fbuffer = f.read(block)
    while len(fbuffer) > 0:
        #hash
        fingerprint.update(fbuffer)
        fbuffer = f.read(block)
    return fingerprint.hexdigest()


# Command Line functionality
if __name__ == "__main__":
    import os
    from argparse import ArgumentParser, FileType
    parser = ArgumentParser(description="Create a hash Fingerprint of a file")
    parser.add_argument("file", help="The file you wish to fingerprint", type=FileType("rb"))
    args = parser.parse_args()
    root = os.getcwd()
    print(fingerprint(args.file))

File: WEBAPP/app/fingerprint.js
var ArgParse, args, crypto, fingerprint, fs, parser, path, src;

crypto = require("crypto");

fs = require("fs");

path = require("path");

fingerprint = function(file, callback) {
  var fd, hash;
  fd = fs.createReadStream(file);
  hash = crypto.createHash("md5");
  hash.setEncoding("hex");
  fd.on("error", function(err) {
    return callback(err, null);
  });
  fd.on("end", function() {
    hash.end();
    return callback(null, hash.read());
  });
  return fd.pipe(hash);
};

ArgParse = require("argparse/lib/argparse").ArgumentParser;

parser = new ArgParse({
  version: "0.0.1",
  addHelp: true,
  description: "Generate a unique hash from a file"
});

parser.addArgument(["source"], {
  help: "The file to be hashed."
});

args = parser.parseArgs();

src = path.resolve(args.source);

fingerprint(src, function(err, hash) {
  console.dir(err);
  return console.dir(hash);
});
